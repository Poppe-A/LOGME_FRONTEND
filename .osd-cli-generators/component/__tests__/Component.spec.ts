const ejs = require('ejs');
const frontmatter = require('frontmatter');

describe('Component generator', () => {
    it('generates correctly', async () => {
        const output = await ejs.renderFile(`${__dirname}/../Component.tsx.ejs`, {
            props: {
                pascalCaseName: 'Foobar',
                usesState: true,
                usesEffect: true,
                isMemoized: true,
            },
        });
        const fmData = frontmatter(output);

        expect(fmData.content).toMatchSnapshot();
    });
});
