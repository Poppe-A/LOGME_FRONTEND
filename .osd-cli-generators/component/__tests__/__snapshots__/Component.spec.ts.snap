// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component generator generates correctly 1`] = `
"import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

const Container = styled.div<IProps>\`
    padding: 1rem;
    border: 1px solid currentColor;
    border-radius: 0.25rem;
\`;

interface IProps {}

const Foobar: React.FC<IProps> = React.memo((props: IProps) => {
    const [state, setState] = useState({});
    useEffect(() => {
        // Do something
    }, []);
    return (
        <Container data-testing=\\"Foobar\\">
            Foobar component!
        </Container>
    );
});

export default Foobar;
"
`;
